### PR TITLE
Advanced character page

### PR DESCRIPTION
**Pull request by Sanae-G**
add router, moved important var and states to App.js and add (defect) toggle function. line 11 to 17 in cardList is not working in this file but toggle works when adding these lines in App.js.

### CODE REVIEW

File: src/components/CardList.jsx [hunk 1]
```diff
@@ -3,20 +3,18 @@ import styled from 'styled-components';
...
           <button onClick={toggleInfo}>Show more</button>
-          <div>
            {showMore? (
-             <>
-            <p>Species: {species}</p>
-            <p>Gender: {gender}</p>
-            <p>Status: {status}</p>
-            </>
+            <ul>
+            <li>Species: hello</li>
```
Comment (ThomasF85) [inline src/components/CardList.jsx]: Why not show the actual values instead of these dummy values?

File: src/components/CardList.jsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ function CardList() {
 
   const apiURL="https://rickandmortyapi.com/api/character";
   const [characters, setCharacters] = useState([]);
+  const [showMore, setShowMore] = useState(false);
```
Comment (ThomasF85) [inline src/components/CardList.jsx]: You have one showMore state for all your cards? What is the intention behind this? This changes the state for all cards even if it is only pressed on one card.

File: src/App.js [hunk 3]
```diff
@@ -6,14 +6,32 @@ import { Routes, Route } from "react-router-dom";
...
+    setShowMore(!showMore);
+  }
+
+
   return (
     <>
       <Header />
       <Routes>
-        <Route path="/" element={<CharacterPage />}/>
+        <Route path="/" element={<CharacterPage characters={characters} toggleInfo={toggleInfo}/>}/>
```
Comment (ThomasF85) [inline src/App.js]: YOu are not giving this component the showMore prop, but the component expects it

File: src/components/CardList.jsx [hunk 4]
```diff
@@ -1,36 +1,14 @@
...
-     return (
-       <CardContent key={id}>
-          <img src={image}>
-          </img>
-         <CharacterName>{name}</CharacterName>
-         <button onClick={addInfo}>Show more</button>
-         <div>
+    <CardListStyled>
+      {characters.map(({ name, image, id, species, gender, status }) => (
+        <CardContent key={id}>
```
Comment (ThomasF85) [inline src/components/CardList.jsx:25]: There should be some kind of link to the details page within the "Card component"

**Review by ThomasF85 — CHANGES_REQUESTED** [review]

File: src/components/CardList.jsx (hunk 2 above)
Comment (Sanae-G) [reply src/components/CardList.jsx]: That is my problem. I do not have another solution. I am struggling with useParams

**Review by ThomasF85 — APPROVED** [review]
Looks good!

**Pull request merged**